### ISSUE TITLE
Help with Camunda's Run distro (Docker)

### ISSUE THREAD

**Issue by julrichkieffer** [StephenOTT/camunda-formio-plugin#51]
Camunda has a self-contained distro called Run. There is also a [docker variant](https://docs.camunda.org/manual/latest/installation/docker/#start-camunda-bpm-run-using-docker) which I'm using.

As I'm a novice to Java, Tomcat, etc, I'm struggling to overlay [your instructions](https://github.com/StephenOTT/camunda-formio-plugin/issues/34) onto Camunda Run. In particular, I'm reluctant to tinker with the `internal` folder as it implies "off limits" :).

Could some instructions to integrate this plugin into Camunda Run (docker variant) be created? Some ideas:
- craft a child `Dockerfile` to build a custom image with this plugin already "deployed"
- use `-v [local file system]:[container file system]` volume mapping(s) to inject this plugin at container start

I use the latter technique to inject my custom .bpmn files into the `/configuration/resources` folder.  However, I've found that only resources in the same deployment (startupDeployment) can be referenced in form keys (`embedded:deployment://`). Whereas I'd prefer a once-off deployment of this plugin be accessible from all processes, even those done after startup (e.g. via Modeler).

**Comment by StephenOTT** [comment]
@julrichkieffer yes I have had a similar request on the forums.  The camunda run Distro uses springboot and there is auto-scanning for components: so we can package the plugin as a jar just for the Run Distro and it will be picked up by Run (all variants) when you drop it into the library folder.

I will look to do it this week.

**Comment by julrichkieffer** [comment]
That's awesome, and thanks for your effort in advance. Might this jar become a release artefact so that any plugin updates are automatically included in the jar?  (I'm particularly keen on [including the FormIO json within the .bpmn file](https://github.com/StephenOTT/camunda-formio-plugin/issues/48), so that it's versioned with the process itself. Though nested forms, validations, boundary events, pre-setting submision values may thwart this approach 🍰).

Also, will this jar simply drop into the `/configuration/userlib` folder (not Tomcat's `lib` folder) to align with Run's design? I would then be able to volume map it when I start Camunda Run as a docker image.

**Comment by StephenOTT** [comment]
yes it would be a jar in the release

see the conversations here: https://forum.camunda.org/t/formio-plugin-for-camunda-tasklist/22215/26  It covers discussion about using forms with the BPMN file, pros, cons, etc.

and yes we can build a example of build a docker container with the Run container and align with it's intended usage.

**Comment by julrichkieffer** [comment]
Any news on a release .jar that would be compatibale with Camunda Run's `/configuration/userlib` extensibility?

**Comment by julrichkieffer** [comment]
> yes it would be a jar in the release
> 
> see the conversations here: https://forum.camunda.org/t/formio-plugin-for-camunda-tasklist/22215/26 It covers discussion about using forms with the BPMN file, pros, cons, etc.
> 
> and yes we can build a example of build a docker container with the Run container and align with it's intended usage.

Any news on a release .jar that would be compatibale with Camunda Run's `/configuration/userlib` extensibility?

#NotAJavaDeveloper #CitizenDeveloper
